source/module_hsolver/src/cuda/math_kernel.cu

```diff
@@ -9,15 +9,20 @@
 namespace hsolver {
 
 
-static cublasHandle_t diag_handle;
+static cublasHandle_t diag_handle = nullptr;
 
 
 void createBLAShandle(){
-    cublasCreate(&diag_handle);
+    if (diag_handle == nullptr) {
+        cublasCreate(&diag_handle);
```

Comment:
Maybe it would be better checking if the return value is normal?

Reply:
LGTM